### PR TITLE
add sd1.5 compatibility to controlnet-xs and fix unused_parameters error during training

### PR DESCRIPTION
**Pull request by SamMaoYS**
# ControlNet-XS SD1.5 compatibility and fix unused_parameters error during training

### Fix unused parameters error during training
The [time_embedding](https://github.com/huggingface/diffusers/blob/eeb70033a6224f58eae3f0f1121384c160a5a61b/src/diffusers/models/controlnet_xs.py#L347) in `ControlNetXSAdapter` is set to None, when not learning the time embedding. But the [ctrl_time_embedding](https://github.com/huggingface/diffusers/blob/eeb70033a6224f58eae3f0f1121384c160a5a61b/src/diffusers/models/controlnet_xs.py#L598C14-L598C33) in `UNetControlNetXSModel` is not set to None, when not learning. This will result in `unused_parameters` when computing the backward gradients during training.

### Add SD1.5 compatibility to ControlNet-XS
SD1.5 set [use_linear_projection](https://github.com/huggingface/diffusers/blob/eeb70033a6224f58eae3f0f1121384c160a5a61b/src/diffusers/models/controlnet_xs.py#L155C21-L155C42) to `False`, but SD2.1 set this parameter to `True`. Instead of hard coding this parameter to `True`, load this parameter from the UNet config instead.

@sayakpaul @yiyixuxu, your time and expertise will be highly appreciated in examining this PR. I am working on ControlNet-XS related research and experiments, hope this PR will make training ControlNet-XS with various SD pipelines easier. Diffusers is an amazing open-source project, I sincerely appreciate everyone's hard work in it, and look forward to contributing my small part to it as well.

### CODE REVIEW

**Comment by sayakpaul** [comment]
Cc: @UmerHA who contributed ControlNet-XS.

File: src/diffusers/models/controlnet_xs.py [hunk 1]
```diff
@@ -1335,13 +1350,16 @@ def get_first_cross_attention(block):
         temb_channels = base_downblock.resnets[0].time_emb_proj.in_features
         num_groups = base_downblock.resnets[0].norm1.num_groups
         ctrl_num_groups = ctrl_downblock.resnets[0].norm1.num_groups
+        use_linear_projection = True # default SD2.1 set linear projection to True
```
Comment (UmerHA) [inline src/diffusers/models/controlnet_xs.py:1353]: tiny suggestion: would not set `use_linear_projection` here, but inside the if/else, and default to `None`.

Same in mid / up block adapter

File: src/diffusers/models/controlnet_xs.py [hunk 2]
```diff
@@ -1335,13 +1350,16 @@ def get_first_cross_attention(block):
...
         if hasattr(base_downblock, "attentions"):
             has_crossattn = True
             transformer_layers_per_block = len(base_downblock.attentions[0].transformer_blocks)
             base_num_attention_heads = get_first_cross_attention(base_downblock).heads
             ctrl_num_attention_heads = get_first_cross_attention(ctrl_downblock).heads
             cross_attention_dim = get_first_cross_attention(base_downblock).cross_attention_dim
             upcast_attention = get_first_cross_attention(base_downblock).upcast_attention
+            if hasattr(base_downblock.attentions[0], "use_linear_projection"):
```
Comment (UmerHA) [inline src/diffusers/models/controlnet_xs.py:1361]: Is there a case where `base_downblock.attentions[0]` doesn't have a `use_linear_projection` attribute?

**Review by UmerHA — APPROVED** [review]
some small comments - overall look's very good to me. Nice!

File: src/diffusers/models/controlnet_xs.py (hunk 1 above)
Comment (SamMaoYS) [reply src/diffusers/models/controlnet_xs.py]: Hi @UmerHA, thank you for the suggestions. I updated the code by following your suggestions, and make `use_linear_projection` more consistent with other parameters.

File: src/diffusers/models/controlnet_xs.py (hunk 2 above)
Comment (SamMaoYS) [reply src/diffusers/models/controlnet_xs.py]: I was considering when `dual_cross_attention` is set to True in `CrossAttnDownBlock2D`, the `DualTransformer2DModel` has no attribute `use_linear_projection `. But I don't think it is a big concern for released SD models. So I updated the code to make it more consistent with other parameters now.

File: src/diffusers/models/controlnet_xs.py (hunk 1 above)
Comment (UmerHA) [reply src/diffusers/models/controlnet_xs.py]: Looks good!

File: src/diffusers/models/controlnet_xs.py (hunk 2 above)
Comment (UmerHA) [reply src/diffusers/models/controlnet_xs.py]: Also looks good!

**Comment by yiyixuxu** [comment]
can we run `make style`?

**Comment by SamMaoYS** [comment]
> can we run `make style`?

Sorry for not running the code style correction in advance, the code is updated with `make style` and `make quality`. Thanks for the suggestions.

**Comment by yiyixuxu** [comment]
thank you!

**Pull request merged**